packages/components/src/range-control/index.js

```diff
@@ -26,19 +27,52 @@ function RangeControl( {
 	help,
 	allowReset,
 	initialPosition,
+	min,
+	max,
+	setState,
 	...props
 } ) {
 	const id = `inspector-range-control-${ instanceId }`;
-	const resetValue = () => onChange();
+	const currentInputValue = currentInput === null ? value : currentInput;
+	const resetValue = () => {
+		resetCurrentInput();
+		onChange();
+	};
+	const resetCurrentInput = () => {
+		// If the input is an empty string call onChange with an undefined argument to reset to value.
```

Comment:
Do we need to call `onChange` at all? If the attempted value wasn't valid, just reset the current state. The rendering parent should still think the value was the "original" since it never received any indication otherwise. Then, `onChange` is only ever called when the input value changes to something valid.

Reply:
My idea was that deleting the input field would reset the field to the default value, in the same way pressing reset button does.
But that is not a requirement and just resetting to the previous value is also valid so I updated the code.